Add docs page for Reanimated Swipeable (#2962)
software-mansion/react-native-gesture-handler: docs/docs/components/reanimated_swipeable.md

method that is called when action panel gets open (either right or left). Takes swipe direction as
an argument.

### `onSwipeableClose`

method that is called when action panel is closed. Takes swipe direction as
an argument.

### `onSwipeableWillOpen`

method that is called when action panel starts animating on open (either right or left). Takes swipe direction as
an argument.

### `onSwipeableWillClose`

method that is called when action panel starts animating on close. Takes swipe direction as
an argument.

### `renderLeftActions`

method that is expected to return an action panel that is going to be revealed from the left side when user swipes right.
This map describes the values to use as inputRange for extra interpolation:
AnimatedValue: [startValue, endValue]

progressAnimatedValue: [0, 1]
dragAnimatedValue: [0, +]

To support `rtl` flexbox layouts use `flexDirection` styling.

### `renderRightActions`

method that is expected to return an action panel that is going to be revealed from the right side when user swipes left.
This map describes the values to use as inputRange for extra interpolation:
AnimatedValue: [startValue, endValue]

progressAnimatedValue: [0, 1]

> dragAnimatedValue: [0, -]

Reviewer: Those are shared values not animated values in the new implementation

Author: replaced those instances

---
id: reanimated_swipeable
title: Reanimated Swipeable
sidebar_label: Reanimated Swipeable
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import GifGallery from '@site/components/GifGallery'

<GifGallery>
  <img src={useBaseUrl("gifs/sampleswipeable.gif")} height="120" />
</GifGallery>

:::info
This component is a drop-in replacement for the `Swipeable` component, rewritten using `Reanimated`.
:::

Reanimated `Swipeable` allows for implementing swipeable rows or similar interaction. It renders its children within a panable container allows for horizontal swiping left and right. While swiping one of two "action" containers can be shown depends on whether user swipes left or right (containers can be rendered by `renderLeftActions` or `renderRightActions` props).

### Usage:

To use it, import it in the following way:

```js
import Swipeable from 'react-native-gesture-handler/ReanimatedSwipeable';
```

## Properties

### `friction`

> a number that specifies how much the visual interaction will be delayed compared to the gesture distance. e.g. value of 1 will indicate that the swipeable panel should exactly follow the gesture, 2 means it is going to be two times "slower".

Reviewer 2: nitpick: any reason to start sentences with small letters rather than capital ones?
rest of the documentation uses capital letters I believe

Author: No reason at all, fixing.

### `leftThreshold`

distance from the left edge at which released panel will animate to the open state (or the open panel will animate into the closed state). By default it's a half of the panel's width.

### `rightThreshold`

distance from the right edge at which released panel will animate to the open state (or the open panel will animate into the closed state). By default it's a half of the panel's width.

### `dragOffsetFromLeftEdge`

> distance that the panel must be dragged from the left edge to be considered a swipe. The default value is 10.

Reviewer 2: ```suggestion
distance that the panel must be dragged from the left edge to be considered a swipe. The default value is `10`.
```
the values should be represented as `code` - it's more readable that way

Author: Changed these, as well as other instances of number values.

### `overshootFriction`

a number that specifies how much the visual interaction will be delayed compared to the gesture distance at overshoot. Default value is 1, it mean no friction, for a native feel, try 8 or above.

### `onSwipeableOpen`

method that is called when action panel gets open (either right or left). Takes swipe direction as
an argument.

### `onSwipeableClose`

method that is called when action panel is closed. Takes swipe direction as
an argument.

### `onSwipeableWillOpen`

method that is called when action panel starts animating on open (either right or left). Takes swipe direction as
an argument.

### `onSwipeableWillClose`

method that is called when action panel starts animating on close. Takes swipe direction as
an argument.

### `renderLeftActions`

method that is expected to return an action panel that is going to be revealed from the left side when user swipes right.
This map describes the values to use as inputRange for extra interpolation:

SharedValue: [startValue, endValue]

progress: [0, 1]

> drag: [0, +]

Reviewer 2: I don't understand the `[0, +]`, `[0, -]` notation

Author: It means these values travel from `0` to `positive infinity` and from `0` to `negative infinity` respectively.
I copied this notation from the legacy `Swipeable` docs for consistency, but i can write it down differently if you'd prefer.

Reviewer 3: `[0, +]` is not that bad, but `[0, -]` is confusing

Author: is `[0, +∞]` and `[0, -∞]` fine?

Reviewer 3: What I meant is not the infinity sign, but rather the notation of `[0, -∞]` (it doesn't follow formal definition of interval). I'm not sure though if that is what @kacperkapusciak meant.

Author: Hey so what's the verdict on this issue?

As far as I see, the formal way would be to note it this way: `(-∞, 0]` and `[0, +∞)`
But since here, it represents a mapping from `input` to `output`, i think `[0, -∞)` and `[0, +∞)` would be more readable.

Please let me know, which one you think would work better @m-bert.

Reviewer 3: Well, I don't like `[0, -∞)` because it looks very unnatural to me. On the other hand I cannot think of any better idea 😞  One could argue that if we only talk about input and output range, we could leave formal notation (but it is not a strong argument)

Anyway, it was @kacperkapusciak who has started this discussion and I want to know what's his stance on this

---
id: reanimated_swipeable
title: Reanimated Swipeable
sidebar_label: Reanimated Swipeable
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import GifGallery from '@site/components/GifGallery'

<GifGallery>
  <img src={useBaseUrl("gifs/sampleswipeable.gif")} height="120" />
</GifGallery>

:::info

> This component is a drop-in replacement for the `Swipeable` component, rewritten in `Reanimated2`.

Reviewer: ```suggestion
This component is a drop-in replacement for the `Swipeable` component, rewritten using `Reanimated`.
```

Author: changed

:::

Reanimated `Swipeable` allows for implementing swipeable rows or similar interaction. It renders its children within a panable container allows for horizontal swiping left and right. While swiping one of two "action" containers can be shown depends on whether user swipes left or right (containers can be rendered by `renderLeftActions` or `renderRightActions` props).

### Usage:

> Similarly to the `DrawerLayout` and `Swipeable`, Reanimated `Swipeable` component isn't exported by default from the `react-native-gesture-handler` package. To use it, import it in the following way:

Reviewer: ```suggestion
To use the component, import it in the following way:
```

Author: changed

### `onSwipeableLeftOpen`

:::caution
This callback is deprecated and will be removed in the next version. Please use `onSwipeableOpen(direction)`
:::

method that is called when left action panel gets open.

### `onSwipeableRightOpen`

:::caution
This callback is deprecated and will be removed in the next version. Please use `onSwipeableOpen(direction)`
:::

method that is called when right action panel gets open.

### `onSwipeableOpen`

method that is called when action panel gets open (either right or left). Takes swipe direction as
an argument.

### `onSwipeableClose`

method that is called when action panel is closed. Takes swipe direction as
an argument.

### `onSwipeableLeftWillOpen`

:::caution
This callback is deprecated and will be removed in the next version. Please use `onSwipeableWillOpen(direction)`
:::

method that is called when left action panel starts animating on open.

### `onSwipeableRightWillOpen`

:::caution
This callback is deprecated and will be removed in the next version. Please use `onSwipeableWillOpen(direction)`
:::

> method that is called when right action panel starts animating on open.

Author: removed

### `mouseButton(value: MouseButton)` (Web & Android only)

Allows users to choose which mouse button should handler respond to. The enum `MouseButton` consists of the following predefined fields:

- `LEFT`
- `RIGHT`
- `MIDDLE`
- `BUTTON_4`
- `BUTTON_5`
- `ALL`

Arguments can be combined using `|` operator, e.g. `mouseButton(MouseButton.LEFT | MouseButton.RIGHT)`. Default value is set to `MouseButton.LEFT`.

### `enableContextMenu(value: boolean)` (Web only)

Specifies whether context menu should be enabled after clicking on underlying view with right mouse button. Default value is set to `false`.

## Methods

Using reference to `Swipeable` it's possible to trigger some actions on it

### `close`

method that closes component.

### `openLeft`

method that opens component on left side.

### `openRight`

method that opens component on right side.

### `reset`

method that resets the swiping states of this `Swipeable` component.

Unlike method `close`, this method does not trigger any animation.

### Example:

> See the [swipeable example](https://github.com/software-mansion/react-native-gesture-handler/blob/main/example/src/showcase/swipeable/index.tsx) from GestureHandler Example App or view it directly on your phone by visiting [our expo demo](https://snack.expo.io/@adamgrzybowski/react-native-gesture-handler-demo).

Author: removed

```js
import React, { Component } from 'react';
import { Animated, StyleSheet, View } from 'react-native';
import { RectButton } from 'react-native-gesture-handler';
import Swipeable from 'react-native-gesture-handler/Swipeable';

class AppleStyleSwipeableRow extends Component {
  renderLeftActions = (progress, dragX) => {
    const trans = dragX.interpolate({
      inputRange: [0, 50, 100, 101],
      outputRange: [-20, 0, 0, 1],
    });
    return (
      <RectButton style={styles.leftAction} onPress={this.close}>
        <Animated.Text
          style={[
            styles.actionText,
            {
              transform: [{ translateX: trans }],
            },
          ]}>
          Archive
        </Animated.Text>
      </RectButton>
    );
  };
  render() {
    return (
      <Swipeable renderLeftActions={this.renderLeftActions}>
        <Text>"hello"</Text>
      </Swipeable>
    );
  }

> }

Author: let me know if the current lengthier example is ok, or if it should be slimmed down a bit more

> }

Reviewer: It looks ok, I haven't tested the code itself.
